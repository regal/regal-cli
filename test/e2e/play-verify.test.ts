import { spawn } from "child_process";
import { execRegal, logbacks } from "../test-utils";

it.skip("Play-Verify", done => {
    const proc = spawn(
        `${execRegal} play ./test/e2e/resources/sample-game/temp/bundle.regal.js`,
        {
            shell: true
        }
    );

    let isDone = false;

    proc.stderr.on("data", data => {
        if (!isDone) {
            isDone = true;
            console.log("stderr");
            // proc.kill();
            done();
        }
    });

    proc.stdout.on("data", data => {
        if (!isDone) {
            console.log("data");
            done();
        }
    });

    while (true) {} // TODO - start here
});
