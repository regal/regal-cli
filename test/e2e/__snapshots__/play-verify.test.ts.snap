// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play-Verify 1`] = `
"Now Playing: Sample Game by Lars
Type :quit to exit the game.

Hello, world!

You just said hey there!.

You've said 1 in total.
Here's a random string for you: AnckGshxhi

You just said foo.
Recently, you said hey there!.
You've said 2 in total.
Here's a random string for you: bIIvZNNnOJ

You just said asdfjf.
Recently, you said foo and hey there!
You've said 3 in total.
Here's a random string for you: nckGshxhiU
"
`;
